http-agent/src/main/java/org/motechproject/http/agent/factory/HttpComponentsClientHttpRequestFactoryWithAuth.java

```diff
@@ -31,14 +34,19 @@ public HttpComponentsClientHttpRequestFactoryWithAuth(String username, String pa
      */
     @Override
     protected HttpContext createHttpContext(HttpMethod httpMethod, URI uri) {
-        return createHttpContext();
+        HttpHost targetHost = new HttpHost(uri.getHost(), uri.getPort(), uri.getScheme());
```

Comment:
We should make this configurable in the properties file, using a property called http.auth.preemptive=true/false